modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySpi.java

```diff
@@ -1686,7 +1687,7 @@ protected void writeToSocket(Socket sock, TcpDiscoveryAbstractMessage msg, byte[
             out.flush();
         }
         catch (IOException e) {
-            err = e;
+            err = maybeSslError(sock, e);
```

Comment:
I suggest to rename this method to `clearUpSslEssence` to better reflect its semantics: find out if exception's context indicates it is an SSL-related error and wrap it into SSLException if needed.